codersdk/deployment.go

```diff
@@ -1834,6 +1835,15 @@ Write out the current server config as YAML to stdout.`,
 			Group:       &deploymentGroupClient,
 			YAML:        "webTerminalRenderer",
 		},
+		{
+			Name:        "Allow Workspace Renames",
+			Description: "DEPRECATED: Allow users to rename their workspaces. Use only for temporary compatibility reasons, this flag will no longer function after 2024-04-01.",
+			Flag:        "allow-workspace-renames",
+			Env:         "CODER_ALLOW_WORKSPACE_RENAMES",
+			Default:     "false",
+			Value:       &c.AllowWorkspaceRenames,
+			YAML:        "allowWorkspaceRenames",
```

Comment:
I considered this, but doing that would break CI on the date it becomes hidden, because the golden files need updating to update the docs. Either way it seems there's gonna be some manual intervention to support it with our current workflows.